File: leetcode/algorithms/c++/Merge Two Sorted Lists/Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *mergeTwoLists(ListNode *l1, ListNode *l2)
    {
        vector<int> vals;
        while (l1 != NULL)
        {
            vals.push_back(l1->val);
            l1 = l1->next;
        }
        while (l2 != NULL)
        {
            vals.push_back(l2->val);
            l2 = l2->next;
        }
        sort(vals.begin(), vals.end());
        if (vals.size() == 0)
            return NULL;

        ListNode *root = new ListNode(vals[0]);
        ListNode *tmp = root;
        for (int i = 1; i < vals.size(); i++)
        {
            tmp->next = new ListNode(vals[i]);
            tmp = tmp->next;
        }
        return root;
    }
};

File: leetcode/algorithms/c#/Merge Two Sorted Lists/Merge Two Sorted Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution
{
    public ListNode MergeTwoLists(ListNode l1, ListNode l2)
    {
        if (l1 == null && l2 == null) return null;

        List<int> l1List = GetList(l1);
        List<int> l2List = GetList(l2);

        List<int> sum = new List<int>();
        sum.AddRange(l1List);
        sum.AddRange(l2List);
        sum.Sort();

        ListNode begin = new ListNode(sum[0]);
        ListNode next = begin;
        for (int i = 1; i < sum.Count; i++)
        {
            next.next = new ListNode(sum[i]);
            next = next.next;
        }
        return begin;


    }

    public List<int> GetList(ListNode x)
    {
        List<int> result = new List<int>();
        while (x != null)
        {
            result.Add(x.val);
            x = x.next;
        }
        return result;
    }
}